src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { ShowType } from './types/ShowType';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [show, setShow] = useState<ShowType>(ShowType.all);
+  const [filter, setFilter] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [showModal, setShowModal] = useState(false);
+  const [loading, setLoading] = useState(true)
+  const [todoError, setTodoError] = useState('')
+
+
+  useEffect(() => {
+    getTodos()
+    .then((value) => {
+      setTodosFromServer(value);
+    })
+    .catch((error) => setTodoError(error.message))
+    .finally(() => setLoading(false))
+
+  }, []);
+
+  const onChangeShow = (value: ShowType) => {
+    setShow(value);
+  };
+
+  const onChangeFilter = (value: string) => {
+    setFilter(value);
+  };
+
+  const onSelectedTodo = (value: Todo | null) => {
+    setSelectedTodo(value);
+  };
+
+  const changeShowModal = (value: boolean) => {
+    setShowModal(value);
+  };
```

Comment:
Not real need in those functions because they doesn't save you anything, just taking extra space

Reply:
removed them all